### PR TITLE
Automated UI Testing - CRUD for Servers, CDNs, Delivery Services

### PR DESCRIPTION
**Pull request by chadgilloth**
#### What does this PR do?
WIP This updates the Protractor UI tests for Login and adds CRUD tests for Servers page, CDNs page and Delivery Services page.

Fixes #(issue_number)

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [x] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?
Go to traffic_portal/test/end_to_end and open the README for instructions on how to install Protractor and run these tests locally.

#### Check all that apply

- [x] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2609/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2627/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2628/
Test PASSed.

**Comment by mitchell852** [comment]
`18 specs, 7 failures
Finished in 63.312 seconds

1) Traffic Portal Login Test Suite should successfully login to Traffic Portal
  Message:
    Failed: script timeout: result was not received in 11 seconds
      (Session info: chrome=69.0.3497.100)
      (Driver info: chromedriver=2.43.600229 (3fae4d0cda5334b4f533bede5a4787f7b832d052),platform=Mac OS X 10.11.6 x86_64)
  Stack:
    ScriptTimeoutError: script timeout: result was not received in 11 seconds

2) Traffic Portal Delivery Services Suite should populate and submit the ds form
  Message:
    Expected false to be true.
  Stack:
    Error: Failed expectation

3) Traffic Portal Delivery Services Suite should update the ds
  Message:
    Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator by.repeater("ds in ::deliveryServices")
  Stack:
    NoSuchElementError: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator by.repeater("ds in ::deliveryServices")

4) Traffic Portal Delivery Services Suite should delete the ds
  Message:
    Failed: No element found using locator: by.buttonText("Delete")
  Stack:
    NoSuchElementError: No element found using locator: by.buttonText("Delete")

5) Traffic Portal Servers Test Suite should fill out form, create button is enabled and submit
  Message:
    Expected false to be true.
  Stack:
    Error: Failed expectation

6) Traffic Portal Servers Test Suite should verify the new Server and then update Server
  Message:
    Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator by.repeater("s in ::servers")
  Stack:
    NoSuchElementError: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator by.repeater("s in ::servers")

7) Traffic Portal Servers Test Suite should delete the new Server
  Message:
    Failed: element not interactable
      (Session info: chrome=69.0.3497.100)
      (Driver info: chromedriver=2.43.600229 (3fae4d0cda5334b4f533bede5a4787f7b832d052),platform=Mac OS X 10.11.6 x86_64)
  Stack:
    ElementNotVisibleError: element not interactable`

**Comment by chadgilloth** [comment]
@mitchell852 Since the tests are linear, if one fails, especially the login, the rest will fail. At any point where in a suite there is a failure, the remaining tests if they depend on a previous test, will fail.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2645/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2646/
Test PASSed.

File: traffic_portal/test/end_to_end/conf.js [hunk 1]
```diff
@@ -18,14 +18,39 @@
...
-    params: {
-        adminUser: 'admin',
-        adminPassword: 'password'
-   }
-}
+	framework: 'jasmine',
+	seleniumAddress: 'http://localhost:4444/wd/hub',
+	baseUrl: 'https://localhost:4443',
+	getPageTimeout: 30000,
+	// specs: [
```
Comment (mitchell852) [inline traffic_portal/test/end_to_end/conf.js]: do u want to get rid of this?

Comment (chadgilloth) [reply traffic_portal/test/end_to_end/conf.js]: Comments removed. Individual tests can be ran using protractor conf.js --suite <specname-spec.js>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2656/
Test PASSed.

**Pull request merged**